draft-ietf-jsonpath-base.md

```diff
@@ -958,19 +961,29 @@ In the resultant nodelist:
 
 Children of an object may occur in any order, since JSON objects are unordered.
 
-### Union Selector
+### List Selector
+
+The list selector allows combining member names, array indices, and
+slices in a single selector.
+
+Note: The list selector was called "union selector" in
+{{JSONPath-orig}}, as it was intended to solve use cases addressed by
+the union selector in XPath.
+However, the term "union" implies the set operation that involves
+merging input sets while avoiding duplicates, so the concept was
+renamed into "list selector".
```

Comment:
I'm not sure this note is necessary for the spec.  I don't think historical context is right for specifying language.  Maybe this kind of information needs its own area in the document or even externally.

Reply:
I believe previous users of JSONPath are part of the target audience.
It would be confusing to just rename the thing.

Reply:
My argument is that there is no historical spec. I think this document should be written as if it is defining the syntax from the start.  Any and all references to previous materials, including XPath, should be in an appendix or a separate document.

Reply:
Previous users of JSONPath did have a spec, just not a very detailed one or a very official one.  Generally, aspects of usability trump arbitrary principles that do not by themselves contribute to a good outcome.  Why do we waste time on this discussion.

Reply:
It was a proposal in a blog post.  Such things need to include this kind of context because users have nothing else to draw upon, and given that XPath was the inspiration, its inclusion in the post is warranted.

However a specification is intended to _define_ a thing.  Purpose is needed, but stories of inspiration are not.  If you want to put some history in an appendix, okay, I guess, but historical context should not exist in the midst of the definition (either syntactical definition or behavioral).

Reply:
I agree with @cabo on this. The term "union" is pretty widespread among JSONPath implementations and the renaming deserves being called out in the body of the document. If we don't explain the rationale, people will wonder why we didn't stick with "union". We could put this in an appendix, but that seems rather artificial/pedantic to me.